source/simulator/checkpoint_restart.cc

```diff
@@ -173,37 +152,147 @@ namespace aspect
 #endif
 
     }
-    pcout << "*** Snapshot created!" << std::endl << std::endl;
-    computing_timer.exit_section();
   }
 
+  template<int dim>
+  void Simulator<dim>::log_checkpoint(const std::string filename_for_triangulation, const std::string filename_for_serialization, bool is_rotating_checkpoint)
+  {
+    std::ofstream checkpoint_log;
+    std::string checkpoint_file_name = parameters.output_directory + checkpoint_log_file;
+
+    if (!Utilities::fexists(checkpoint_file_name))
+      {
+        checkpoint_log.open(checkpoint_file_name, std::ios_base::out);
+        checkpoint_log << "Time_step_number filename_triangulation filename_serialized_data rotating_checkpoint_slot_id" << std::endl;
+      }
+    else
+      checkpoint_log.open(checkpoint_file_name, std::ios_base::app);
```

Comment:
You can save yourself some lines of code (and complicated to read code flow) if you write it like this:
```
  const bool checkpoint_file_exists = Utilities::fexists(checkpoint_file_name);
  std::ofstream checkpoint_log (checkpoint_file_name, std::ios_base::app);

  // if file did not exist before, write a preamble into it:
  checkpoint_log << "Time_step_number filename_triangulation filename_serialized_data rotating_checkpoint_slot_id" << std::endl;
```